tools/diffscripts/README.md

```diff
@@ -0,0 +1,208 @@
+# CSharp SDK Diff Automation
+
+## Overview
+During the release process we need to examine the changes to the SDK surface to make sure we're not making too many changes, breaking or otherwise. This tool will automate the process of generating the mark down files so we do not need to use the GUI version which is time consuming.
+
+
+## Quickstart
+This quickstart assumes that you clone your repositories to `c:\repos` and that you've already cloned the CSharp SDK to `c:\repos\azure-iot-sdk-csharp`
+
+```ps
+cd c:\repos
+git clone https://github.com/Azure/iot-sdks-internals.git
+dotnet tool install Microsoft.Dotnet.AsmDiff -g --add-source https://pkgs.dev.azure.com/dnceng/public/_packaging/dotnet-eng/nuget/v3/index.json --version 6.0.0-beta.21161.15
+cd c:\repos\azure-iot-sdk-csharp\tools\difftools
+.\diffapi.ps1
+```
+
+# Prerequisites
+
+## git
+You most likely have this if you're working with a cloned repository.
+
+### Links
+[git](https://git-scm.com/download/win)
+
+## Cloned SDK Internals Repository
+In order to make the comparisons you will need to clone the [internals repo](https://github.com/Azure/iot-sdks-internals) to a location on your local machine.
+
+## dotnet core SDK (>=2.1)
+You should already have this as part of your dev build since we use this to build our sources. However you can download them from the following locations.
+
+### Links
+
+[dotnetcore LTS](https://dotnet.microsoft.com/download/dotnet/thank-you/sdk-3.1.407-windows-x64-installer)
+
+## AsmDiff
+This is the tool that is used to create the mark down and it is written by the dotnet team and is located in the [dotnet/arcade](https://github.com/dotnet/arcade) repository. To install this tool run the following:
+
+```
```

Comment:
Do we need to run powershell in admin mode at all for this? If yes, might be good to add a check and throw an error in the beginning itself so the user knows to run in admin mode.

Reply:
I like that, but luckily we do not need to run as admin. All of my testing has been done in non-admin mode. As well installation of the dotnet tools does not require admin.